### PR TITLE
feat(1109):  opc-nav's search component partitoned to another component 

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Fixes

1109

# Explain the feature/fix

1. Implemented opc-nav-search component for opc-nav, by decoupling it from  its search component. This helps keeping the component atomic and also ease to control the search action.
2. Added another property `activeButton` to control which default nav button is active.
3. Updated test and added new one for opc-nav-search.
4. Updated documentation.

## Does this PR introduce a breaking change

Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

Screenshot 2021-08-05 at 10 07 06 PM

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [x] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [x] Does your component follow One Platform style guidelines?
- [x] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [x] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [ ] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [x] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

File: packages/opc-nav/README.md [hunk 1]
```diff
@@ -80,13 +82,11 @@ document.querySelector('opc-nav').links = links;
...
 <details>
-<summary>opc-nav with search and nav buttons</summary>
+<summary>opc-nav with nav buttons</summary>
 
 ## Details
 
-The search bar component helps users to search for information. It is customizable by using the slot `opc-nav-search` . It can be removed by the attribute `isSearchHidden` with the value true.
-
-The nav buttons, enable utility actions that enhance the user experience. By default, opc-nav provids two buttons with there corresponding events. Menu buttons can be replaced with the slot `opc-nav-btn`. Icon buttons with size of 60px are prefered.
+The nav buttons, enable utility actions that enhance the user experience. By default, opc-nav provids two buttons with there corresponding events. Menu buttons can be replaced with the slot `opc-nav-btn`. Icon buttons with size of 60px are prefered. `activeMenu` property provides active class styling to show users which one is selected. It accepts `menu | notification` as value.
```
Comment (riginoommen) [inline packages/opc-nav/README.md:89]:
```suggestion
The nav buttons, enable utility actions that enhance the user experience. By default, opc-nav provides two buttons with there corresponding events. Menu buttons can be replaced with the slot `opc-nav-btn`. Icon buttons with size of 60px are prefered. `activeMenu` property provides active class styling to show users which one is selected. It accepts `menu | notification` as value.
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/opc-nav/README.md (hunk 1 above)
Comment (akhilmhdh) [reply packages/opc-nav/README.md]: Done

File: packages/opc-nav/src/opc-nav.ts [hunk 2]
```diff
@@ -114,3 +72,108 @@ export class OpcNav extends LitElement {
...
+              ? 'opc-nav-search__input-active'
+              : ''}"
+            type="search"
+            name="query"
+            autofocus
+            autocomplete="off"
+            aria-label="Search"
+            placeholder="Search"
+            required
+            .value=${live(this.value)}
```
Comment (ghanlohar) [inline packages/opc-nav/src/opc-nav.ts:171]: @akhilmhdh should it be without dot.
```suggestion
            value=${live(this.value)}
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/opc-nav/src/opc-nav.ts (hunk 2 above)
Comment (akhilmhdh) [reply packages/opc-nav/src/opc-nav.ts:171]: https://lit.dev/docs/templates/directives/#live 
I implemented this with help of live directive documentation. They are adding . with template attribute even if string in example. I think its due to using directive

**Review by riginoommen — APPROVED** [review]

**Review by ghanlohar — APPROVED** [review]
LGTM

**Comment by hybridx** [comment]
@akhilmhdh I see that there's a misalignment between the nav title and the nav elements, can you please address that?

**Review by hybridx — APPROVED** [review]

**Pull request merged**